cdragontoolbox/export.py

```diff
@@ -175,9 +176,9 @@ def export(self):
         """
 
         if self.previous_patch:
-            logger.info("exporting patch %s based on patch %s", self.patch.version, self.previous_patch.version)
```

Comment:
There is a reason for using `logger.info(fmt, arg1, arg2)` instead of a format string.
The message will only be formatted if the log is actually handled and output. For instance, debug messages will not be formatted if debug messages are dropped (which is the default).
Also, the logging API allows to define custom handlers, LogRecord classes, etc. that could make use of the args being separated.

Even if in our case the only would be a (negligeable) performance impact, I think it's good habit to keep the args separated.
What's your opinion?

Reply:
This is kinda a tough one, but I think that the formatting of a string is such an insignificant time contribution to this code in particular that we shouldn't worry about it. f-strings are just so nice -- much much nicer than the `%` format, and it looks weird to be mixing old and new formatting options (I changed some of them then decided on this).

It would be different if the arguments were not evaluated unless the logger was used, but they have to be evaluated to read the "outer" method scope.

If people are using complex logging via custom handlers etc, then imo they can either submit a PR with any changes they need to just fork the repo and make their changes there. I think it's very unlikely that anyone is going to be using this code as-is in an app that requires customizable logging.

Reply:
Resolved